src/System.Linq.Parallel/tests/QueryOperators/AllTests.cs

```diff
@@ -0,0 +1,113 @@
+
+        //
+        // All
+        //
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void All_AllFalse(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            Assert.Equal(count == 0, query.All(x => x < 0));
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void All_AllFalse_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            All_AllFalse(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("Ranges", (object)(new int[] { 0, 1, 2, 16 }), MemberType = typeof(UnorderedSources))]
+        public static void All_AllTrue(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            IntegerRangeSet seen = new IntegerRangeSet(0, count);
+            Assert.True(query.All(x => seen.Add(x)));
+            Assert.True(seen.Complete());
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("Ranges", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }), MemberType = typeof(UnorderedSources))]
+        public static void All_AllTrue_Longrunning(Labelled<ParallelQuery<int>> labelled, int count)
+        {
+            All_AllTrue(labelled, count);
+        }
+
+        [Theory]
+        [MemberData("OnlyOneData", (object)(new int[] { 2, 16 }))]
+        public static void All_OneFalse(Labelled<ParallelQuery<int>> labelled, int count, int position)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            Assert.False(query.All(x => !(x == position)));
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("OnlyOneData", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }))]
+        public static void All_OneFalse_Longrunning(Labelled<ParallelQuery<int>> labelled, int count, int position)
+        {
+            All_OneFalse(labelled, count, position);
+        }
+
+        [Theory]
+        [MemberData("OnlyOneData", (object)(new int[] { 2, 16 }))]
+        public static void All_OneTrue(Labelled<ParallelQuery<int>> labelled, int count, int position)
+        {
+            ParallelQuery<int> query = labelled.Item;
+            Assert.False(query.All(x => x == position));
+        }
+
+        [Theory]
+        [OuterLoop]
+        [MemberData("OnlyOneData", (object)(new int[] { 1024 * 1024, 1024 * 1024 * 4 }))]
+        public static void All_OneTrue_Longrunning(Labelled<ParallelQuery<int>> labelled, int count, int position)
+        {
+            All_OneTrue(labelled, count, position);
+        }
+
+        [Fact]
+        public static void All_OperationCancelledException()
+        {
+            CancellationTokenSource cs = new CancellationTokenSource();
```

Comment:
Hmm... for the combinatorial, I generally did only one of the 'outer' functions each, just as a SAN check things worked in combination.  I don't test only one item being true in the combinatorial (just all/none).